File: Domain/Entities/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Vehicle: BaseEntity
    {
        public string? SerialNumber { get; set; }
        public int ReleaseYear { get; set; }
        public long Km { get; set; }
        public int VehicleModelId { get; set; }
        public VehicleModel? Model { get; set; }
        public int ClientId { get; set; }
        public Client? Client { get; set; }
        public int FuelTypeId { get; set; }
        public FuelType? FuelType { get; set; }
        public int VehicleTypeId { get; set; }
        public VehicleType? vehicleType { get; set; }
        public List<ServiceOrder>? ServiceOrders { get; set; } = new List<ServiceOrder>();
        public List<VehicleAnormalityDetail>? VehicleAnormalityDetails { get; set; } = new List<VehicleAnormalityDetail>();
    }
}

File: autotaller-manager/src/app/modules/admin/models/vehicle.ts
import { VehicleModel } from './vehicle-model';
import { Client } from './client';
import { FuelType } from './fuel-type';
import { VehicleType } from './vehicle-type';

export interface Vehicle {
  id: number;
  serialNumber: string;
  releaseYear: number;
  km: number;
  vehicleModelId: number;
  model?: VehicleModel;
  clientId: number;
  client?: Client;
  fuelTypeId: number;
  fuelType?: FuelType;
  vehicleTypeId: number;
  vehicleType?: VehicleType;
    
  createdAt?: Date;
  updatedAt?: Date;}
